src/biosHandler.cpp

```diff
@@ -0,0 +1,50 @@
+#include "biosHandler.hpp"
+
+#include "logger.hpp"
+
+namespace vpd
+{
+void BiosHandler::checkAndListenPldmService()
+{
+    // Setup a call back match on NameOwnerChanged to determine when PLDM is up.
+    static std::shared_ptr<sdbusplus::bus::match_t> l_nameOwnerMatch =
+        std::make_shared<sdbusplus::bus::match_t>(
+            *m_asioConn,
+            sdbusplus::bus::match::rules::nameOwnerChanged(
+                "xyz.openbmc_project.PLDM"),
+            [this](sdbusplus::message_t& l_msg) {
+        if (l_msg.is_method_error())
+        {
+            logging::logMessage(
+                "Error in reading PLDM name owner changed signal.");
+            return;
+        }
+
+        std::string l_name;
+        std::string l_newOwner;
+        std::string l_oldOwner;
+
+        msg.read(l_name, l_oldOwner, l_newOwner);
+        if (!l_newOwner.empty() && l_name == "xyz.openbmc_project.PLDM")
```

Comment:
can we use string's compare() function here ?

Reply:
Done.